platform/windows/os_windows.cpp

```diff
@@ -1324,13 +1324,16 @@ void OS_Windows::unset_environment(const String &p_var) const {
 }
 
 String OS_Windows::get_stdin_string() {
-	WCHAR buff[1024];
-	DWORD count = 0;
-	if (ReadConsoleW(GetStdHandle(STD_INPUT_HANDLE), buff, 1024, &count, nullptr)) {
-		return String::utf16((const char16_t *)buff, count);
-	}
-
-	return String();
+    LocalVector<char> buff;
+    DWORD size = 0, count = 0;
+    HANDLE pipe = GetStdHandle(STD_INPUT_HANDLE);
+    if (PeekNamedPipe(pipe, nullptr, 0, nullptr, &size, nullptr)) {
```

Comment:
This assumes `stdin` is a pipe, but is it a `pipe` when redirected to read from file (with ` < file`)?

Reply:
I'll try that now. Will get back to you once I have results.

Reply:
It has been confirmed my code does not work with `executable < file`. However my code is also more correct with what is currently in Godot, as what you currently have doesn't even work with `echo text | executable`, however my code does work with `echo text | executable`. Something that partially works is better than something that doesn't work at all. not to mention you can do this in command prompt as a workaround, (though there still seems to be a character limit with this approach, it's not as small as 1024): `type file | executable`.